src/services/hive/hive.service.ts

```diff
@@ -92,64 +99,84 @@ export class HiveService {
         hiveUsername,
         process.env.ACCOUNT_CREATOR,
       );
-      await this.#linkedAccountsRepository.linkHiveAccount(sub, hiveUsername);
+      await this.#legacyHiveAccountRepository.insertCreated({ account: hiveUsername, user_id });
       return accountWithAuthority;
     } catch (ex) {
       throw new HttpException({ reason: `On chain error - ${ex.message}` }, HttpStatus.BAD_REQUEST);
     }
   }
 
-  async insertCreated(hiveUsername: string, sub: string) {
-    await this.#hiveAccountRepository.insertCreated(hiveUsername, sub);
+  async insertCreated(hiveUsername: string, user_id: ObjectId) {
+    await this.#legacyHiveAccountRepository.insertCreated({ account: hiveUsername, user_id });
   }
 
-  async linkHiveAccount(sub: string, hiveUsername: string, proof: string): Promise<LinkedAccount> {
-    const linkedAccount = await this.#linkedAccountsRepository.findOneByUserIdAndAccountName({
-      user_id: sub,
-      account: hiveUsername,
-    });
+  async linkHiveAccount({
+    db_user_id,
+    user_id,
+    hiveUsername,
+    proof,
+  }: {
+    db_user_id: ObjectId;
+    user_id: string;
+    hiveUsername: string;
+    proof: string;
+  }) {
+    const linkedAccount =
+      await this.#legacyHiveAccountRepository.findOneByOwnerIdAndHiveAccountName({
+        user_id: db_user_id,
+        account: hiveUsername,
+      });
     if (linkedAccount) {
       throw new HttpException({ reason: 'Hive account already linked' }, HttpStatus.BAD_REQUEST);
     }
     const hiveAccount = await this.#hiveChainRepository.getAccount(hiveUsername);
     if (!hiveAccount)
       throw new NotFoundException(`Requested hive account (${hiveAccount}) could not be found.`);
     await this.#hiveChainRepository.verifyHiveMessage(
-      `${sub} is the owner of @${hiveUsername}`,
+      `${user_id} is the owner of @${hiveUsername}`,
       proof,
       hiveAccount,
     );
-    return (await this.#linkedAccountsRepository.linkHiveAccount(
-      sub,
-      hiveUsername,
-    )) satisfies LinkedAccount;
+    return await this.#legacyHiveAccountRepository.insertCreated({
+      account: hiveUsername,
+      user_id: db_user_id,
+    });
   }
 
-  async isHiveAccountLinked(sub: string, accountName: string) {
-    return !!(await this.#linkedAccountsRepository.findOneByUserIdAndAccountName({
-      user_id: sub,
-      account: accountName,
+  async isHiveAccountLinked({ user_id, account }: { user_id: ObjectId; account: string }) {
+    return !!(await this.#legacyHiveAccountRepository.findOneByOwnerIdAndHiveAccountName({
+      user_id,
+      account,
     }));
   }
 
-  async subAuthorizedToUseHiveAccount({
+  async authorizedToUseHiveAccount({
     hiveAccount,
+    user_id,
     sub,
   }: {
     hiveAccount: string;
-    sub: string;
+    user_id: string;
+    sub?: string;
```

Comment:
Ah yes the tech debt. Probably fine for now, but probably should be removed in the future I think.

Reply:
What do you mean? we use this further down to work out if they have authority over an account